### ISSUE TITLE
Running make on a release tag (2.0.1) fails

### ISSUE THREAD

**Issue by ganeshmaharaj** [kata-containers/kata-containers#1558]
# Description of problem
Attempts to build a released tag `2.0.1` on a brand new machine with latest version of rust installed fails to compile.

# Expected result
All kata related binaries are compiled without errors and binaries available for consumption

# Actual result
```
    |
481 |     Ok(AgentServiceClient::new(ttrpc_client))
    |                                ^^^^^^^^^^^^ expected struct `ttrpc::client::Client`, found struct `Client`
    |
    = note: perhaps two different versions of crate `ttrpc` are being used?

error[E0308]: mismatched types
   --> src/client.rs:487:26
    |
487 |     Ok(HealthClient::new(ttrpc_client))
    |                          ^^^^^^^^^^^^ expected struct `ttrpc::client::Client`, found struct `Client`
    |
    = note: perhaps two different versions of crate `ttrpc` are being used?

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0308`.
error: could not compile `kata-agent-ctl`

Caused by:
  process didn't exit successfully: `rustc --crate-name kata_agent_ctl --edition=2018 src/main.rs --error-format=json --json=diagnostic-rendered-ansi --crate-type bin --emit=dep-info,link -C embed-bitcode=no -C debuginfo=2 -C metadata=b60ee8a5c6764f3b -C extra-filename=-b60ee8a5c6764f3b --out-dir /home/ganeshma/kata-containers/tools/agent-ctl/target/debug/deps -C incremental=/home/ganeshma/kata-containers/tools/agent-ctl/target/debug/incremental -L dependency=/home/ganeshma/kata-containers/tools/agent-ctl/target/debug/deps --extern anyhow=/home/ganeshma/kata-containers/tools/agent-ctl/target/debug/deps/libanyhow-554a9d44282dce15.rlib --extern byteorder=/home/ganeshma/kata-containers/tools/agent-ctl/target/debug/deps/libbyteorder-ae594f6f1a6ba442.rlib --extern clap=/home/ganeshma/kata-containers/tools/agent-ctl/target/debug/deps/libclap-13446abce3b7929a.rlib --extern hex=/home/ganeshma/kata-containers/tools/agent-ctl/target/debug/deps/libhex-4df68989c10a4f44.rlib --extern humantime=/home/ganeshma/kata-containers/tools/agent-ctl/target/debug/deps/libhumantime-af024cc6dcec003a.rlib --extern lazy_static=/home/ganeshma/kata-containers/tools/agent-ctl/target/debug/deps/liblazy_static-1fa5e8b52f04e552.rlib --extern libc=/home/ganeshma/kata-containers/tools/agent-ctl/target/debug/deps/liblibc-8b9da7579e43435d.rlib --extern logging=/home/ganeshma/kata-containers/tools/agent-ctl/target/debug/deps/liblogging-8e776fda0bbe1ddb.rlib --extern nix=/home/ganeshma/kata-containers/tools/agent-ctl/target/debug/deps/libnix-ff25ba72fe09fdb4.rlib --extern oci=/home/ganeshma/kata-containers/tools/agent-ctl/target/debug/deps/liboci-c6acb91720bc5b35.rlib --extern protobuf=/home/ganeshma/kata-containers/tools/agent-ctl/target/debug/deps/libprotobuf-e1f1b65e2aa52e0c.rlib --extern protocols=/home/ganeshma/kata-containers/tools/agent-ctl/target/debug/deps/libprotocols-482046919b756ce9.rlib --extern rand=/home/ganeshma/kata-containers/tools/agent-ctl/target/debug/deps/librand-3d03213bf6e8651f.rlib --extern rustjail=/home/ganeshma/kata-containers/tools/agent-ctl/target/debug/deps/librustjail-bd588c48150d7b0a.rlib --extern serde=/home/ganeshma/kata-containers/tools/agent-ctl/target/debug/deps/libserde-12cf2aff4e4c622c.rlib --extern serde_json=/home/ganeshma/kata-containers/tools/agent-ctl/target/debug/deps/libserde_json-6d7e741c04e4a795.rlib --extern slog=/home/ganeshma/kata-containers/tools/agent-ctl/target/debug/deps/libslog-f2cbc881cc480690.rlib --extern slog_scope=/home/ganeshma/kata-containers/tools/agent-ctl/target/debug/deps/libslog_scope-eb302e61c10e5be9.rlib --extern ttrpc=/home/ganeshma/kata-containers/tools/agent-ctl/target/debug/deps/libttrpc-7f8fcc3026d54319.rlib --deny warnings` (exit code: 1)
make[1]: *** [Makefile:9: build] Error 101
```
# Further information
Is there a way to restrict crate updates by rust when building older versions of the code similar to golang's vendored code?

Reproduction steps
1. New linux machine (distro of your liking)
2. Install make, golang, g++
3. Install rustup `curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh`
4. `rustup target add "${arch}-unknown-linux-musl"`
5. `sudo ln -s /usr/bin/g++ /bin/musl-g++`
6. `git checkout 2.0.1`
7. `make -j`

**Comment by egernst** [comment]
hola amigo. what version of rust you running?

**Comment by ganeshmaharaj** [comment]
@egernst  howdy senor. :wink:  I installed the latest stable version, in this case it was 1.50.0.

**Comment by jodh-intel** [comment]
Hi @ganeshmaharaj - I think you just hit https://github.com/kata-containers/kata-containers/pull/1472.

**Comment by ganeshmaharaj** [comment]
@jodh-intel  Seems like it. Is this now a candidate to backport? I doubt we can fix `2.0.1` and `2.0.2` but the next release would be good to be covered.

**Comment by fidencio** [comment]
Let me close this one as 2.0.3 was released already.
